src/main/java/co/cask/gcp/common/StructuredToOrcTransformer.java

```diff
@@ -0,0 +1,120 @@
+
+    @Override
+    public OrcStruct transform(StructuredRecord input, Schema schema) {
+        List<Schema.Field> fields = input.getSchema().getFields();
+        OrcStruct orcRecord = parseOrcSchema(input.getSchema());
+        // populate ORC struct orcRecord object
+        for (int i = 0; i < fields.size(); i++) {
+            Schema.Field field = fields.get(i);
+            try {
+                WritableComparable writable = convertToWritable(field, input);
+                orcRecord.setFieldValue(field.getName(), writable);
+            } catch (UnsupportedTypeException e) {
+                LOG.debug("{} is not a supported type", field.getName(), e);
+                throw new IllegalArgumentException(String.format("%s is not a supported type", field.getName()), e);
+            }
+        }
+        return orcRecord;
+    }
+
+    private OrcStruct parseOrcSchema(Schema inputSchema) {
+        OrcStruct orcRecord;
+        if (schemaCache.containsKey(inputSchema)) {
+            TypeDescription schema = schemaCache.get(inputSchema);
+            orcRecord = (OrcStruct) OrcStruct.createValue(schema);
+        } else {
+            StringBuilder builder = new StringBuilder();
+            try {
+                HiveSchemaConverter.appendType(builder, inputSchema);
+            } catch (UnsupportedTypeException e) {
+                throw new IllegalArgumentException(String.format("Not a valid Schema %s", inputSchema.toString()), e);
+            }
+            TypeDescription schema = TypeDescription.fromString(builder.toString());
+            orcRecord = (OrcStruct) OrcStruct.createValue(schema);
+            schemaCache.put(inputSchema, schema);
+        }
+        return orcRecord;
+    }
+
+    private WritableComparable convertToWritable(Schema.Field field, StructuredRecord input)
+            throws UnsupportedTypeException {
+        Object fieldVal = input.get(field.getName());
+        Schema fieldSchema = field.getSchema();
+        Schema.Type fieldType = fieldSchema.getType();
+        if (fieldSchema.isNullable()) {
+            if (fieldVal == null) {
+                return null;
+            }
+            fieldType = fieldSchema.getNonNullable().getType();
+        }
+        switch (fieldType) {
+            case NULL:
+                return null;
+            case STRING:
+                return new Text((String) fieldVal);
+            case ENUM:
+                return new Text(fieldVal.toString());
+            case BOOLEAN:
+                return new BooleanWritable((Boolean) fieldVal);
+            case INT:
+                return new IntWritable((Integer) fieldVal);
+            case LONG:
+                return new LongWritable((Long) fieldVal);
+            case FLOAT:
+                return new FloatWritable((Float) fieldVal);
+            case DOUBLE:
+                return new DoubleWritable((Double) fieldVal);
+            case BYTES:
```

Comment:
Could you please include Timestamp, Datetime also. I think they are important from DB perspective.